### PR TITLE
[4.0] add fa Regular icons back into core availability

### PR DESCRIPTION
**Pull request by N6REJ**
Pull Request for Issue # .

### Summary of Changes
adds back the ability to use fontawesome regular fonts into core.

### Testing Instructions
go to /administrator/index.php
inspect eye.
do the same for front end login
apply pr
using the inspector, change "fas fa-eye fa-fw" to "far fa-eye fa-fw"
notice the icon changed.

### Actual result BEFORE applying this Pull Request
image

### Expected result AFTER applying this Pull Request
image

### Documentation Changes Required
Our implementation of "Fontawesome Regular icons" is a limited 151 icon set from the 1852 icons included in their "pro" version.  This is by design from FA.  They don't mention that ANY regular icons are included in the free set at all on their site.  Only the icons listed [here](https://fontawesome.com/icons?d=gallery&s=regular&m=free) are included.

### CODE REVIEW

File: build/build-modules-js/settings.json [hunk 1]
```diff
@@ -255,7 +255,9 @@
         },
         "filesExtra": {
           "scss": "scss",
-          "webfonts": "webfonts"
+          "webfonts": "webfonts",
+          "sprites": "sprites",
+          "svgs": "images"
```
Comment (richard67) [inline build/build-modules-js/settings.json]: Is this change intended to be made with this PR here? You have another PR #31079 which does this change. And it is being discussed there.

Is this here now by accident because you created the branch for this PR here based on the branch of the other PR?

Comment (sandewt) [reply build/build-modules-js/settings.json]: PR #31079 would be better now for _joomla:4.1-dev_ and Not for _joomla:4.0-dev_.

Comment (richard67) [reply build/build-modules-js/settings.json]: @sandewt That has (almost) nothing to do with my question above why this change is in this PR here.

Comment (sandewt) [reply build/build-modules-js/settings.json]: See https://github.com/joomla/joomla-cms/issues/30193#issuecomment-708890853

Comment (richard67) [reply build/build-modules-js/settings.json]: This doesn't change the fact that right now there are 2 PRs with this change. It should not be in this PR here.

Comment (sandewt) [reply build/build-modules-js/settings.json]: Correct.

**Comment by adj9** [comment]
I have tested this item :white_check_mark: successfully on 8a17b0382d3e6def550a6b4a0452f3afe0c7f2f9<br /><br />There is the modify in HTML.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31096">issues.joomla.org/tracker/joomla-cms/31096</a>.</sub>

**Comment by richard67** [comment]
@adj9 How can you mark a PR with a successful test when there is a discussion on it about an obvious mistake in it?

Maybe the discussion is not visible in the issue tracker because it was a code review comment. But please always check a PR on GitHub, too, not only in the issue tracker, because only on GitHub you can see which files were changed and if there are unresolved review comments.

Thanks in advance.

File: build/build-modules-js/settings.json (hunk 1 above)
Comment (N6REJ) [reply build/build-modules-js/settings.json]: gosh damn it!

**Comment by richard67** [comment]
@adj9 Please repeat your test. Thanks in advance.

**Comment by ceford** [comment]
I tested this but did not see any change in the source code and the diff only shows changes to files processed by npm.

**Comment by richard67** [comment]
@ceford As far as I understood the testing instructions, you will only see a difference if you change the code in the html markup from "fas fa-eye fa-fw" to "far fa-eye fa-fw" e.g. by manipulating the markup in your browser's developer tools or by editing the PHP file. The icon should then visibly change it's look from solid to regular.

The changes of this PR here do not change the html markup, they only change the scss and so the compiled css so that the icon still is shown when far is used instead of fas.

@N6REJ Correct me if my understanding is wrong.

**Comment by ceford** [comment]
Baffled again! I now see the eye icon change appearance in atum. But it does not make any difference whether or not I have the patch applied. This last part of the dif looks odd:
<pre>
 // Backend Template stuff
 @import "../../../administrator/templates/atum/scss/mixin";
diff --git a/templates/cassiopeia/scss/template.scss b/templates/cassiopeia/scss/template.scss
index 896fb521bf09..1544b8a27a1f 100644
--- a/templates/cassiopeia/scss/template.scss
+++ b/templates/cassiopeia/scss/template.scss
@@ -14,8 +14,9 @@
 
 // "Font Awesome 5 Free"
 @import "../../../media/vendor/fontawesome-free/scss/fontawesome";
-@import "../../../media/vendor/fontawesome-free/scss/solid";
 @import "../../../media/vendor/fontawesome-free/scss/brands";
+@import "../../../media/vendor/fontawesome-free/scss/regular";
+@import "../../../media/vendor/fontawesome-free/scss/solid";
 </pre>

There is no mention of import into atum.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31096">issues.joomla.org/tracker/joomla-cms/31096</a>.</sub>

**Comment by N6REJ** [comment]
@richard67 100% on the money
@ceford for some reason it was never removed in atum ( which turned out to be a good thing ) https://github.com/joomla/joomla-cms/blob/5813af58abdc61c8c71bcdf4302d5a0bb1b505b3/administrator/templates/atum/scss/vendor/fontawesome-free/fontawesome.scss#L6
the diff probably looks odd to you because I put the imports in alphabetical order.
which brings up that atum was NOT in proper order at all.  So thanks for that.

**Comment by particthistle** [comment]
I have tested this item :white_check_mark: successfully on fb5cc3ebb0557d279b94e9802451ae4fb0c0f111<br /><br />Swapping fas for far when inspecting code on the administrator area after applying the patch changed the icon from solid to regular as expected as the result.

Handy to know Font Awesome's Regular free icons are now included in the backend.

@N6REJ can you replicate this PR over into Cassiopeia, as seeing if it carried over to the front end didn't repeat the backend behaviour.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31096">issues.joomla.org/tracker/joomla-cms/31096</a>.</sub>

**Comment by N6REJ** [comment]
@particthistle not sure how to do that but I can try.

**Comment by particthistle** [comment]
> @particthistle not sure how to do that but I can try.

Looking at it more this afternoon it might be a little more difficult as the fonts are currently in the atum template, and not a library loading on it's own.

As mentioned though in #30793 once you know where the files are you could reference them when building a template.

**Comment by degobbis** [comment]
I have tested this item :red_circle: unsuccessfully on fb5cc3ebb0557d279b94e9802451ae4fb0c0f111<br /><br />Tested unsuccessfully with 4.0-dev after run ```npm run build:css```<br />The icon ```far fa-eye fa-fw``` looks equal before and after the patch.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31096">issues.joomla.org/tracker/joomla-cms/31096</a>.</sub>

**Comment by particthistle** [comment]
@degobbis The PR doesn't actually change the class to far fa-eye fa-fw, you need to do that with a manual code inspection.

Here's what the login page fa-eye looks like after my test...

**Comment by degobbis** [comment]
@particthistle This is exactly what I did, before and after the patch.
The icon `far fa-eye fa-fw` looks the same before and after the patch.

**Comment by N6REJ** [comment]
> @particthistle This is exactly what I did, before and after the patch.
> The icon `far fa-eye fa-fw` looks the same before and after the patch.

it's a VERY subtle change. if you want something more dramatic a shift use envelope.. https://fontawesome.com/icons?d=gallery&q=envelope&m=free
so, 'far fa-envelope fa-fw" & "fas fa-envelope fa-fw"

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on fb5cc3ebb0557d279b94e9802451ae4fb0c0f111<br /><br />Confirmed after patch that changing to "far fa-eye fa-fw" on eye image results in the new eye icon. This was confirmed on front end, backend, and installation.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31096">issues.joomla.org/tracker/joomla-cms/31096</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31096">issues.joomla.org/tracker/joomla-cms/31096</a>.</sub>
